src/main/java/htsjdk/variant/vcf/VCFPathReader.java

```diff
@@ -0,0 +1,183 @@
+        this.reader = AbstractFeatureReader.getFeatureReader(
+                path.toUri().toString(),
+                codec,
+                requireIndex);
+    }
+
+    /**
+     * Allows construction of a VCFFileReader with a specified index path.
+     */
+    public VCFPathReader(final Path path, final Path indexPath, final boolean requireIndex) {
+        // Note how we deal with type safety here, just casting to (FeatureCodec)
+        // in the call to getFeatureReader is not enough for Java 8.
+        FeatureCodec<VariantContext, ?> codec = isBCF(path) ? new BCF2Codec() : new VCFCodec();
+        this.reader = AbstractFeatureReader.getFeatureReader(
+                path.toUri().toString(),
+                indexPath.toUri().toString(),
+                codec,
+                requireIndex);
+    }
+
+
+    /**
+     * Parse a VCF file and convert to an IntervalList The name field of the IntervalList is taken from the ID field of the variant, if it exists. if not,
+     * creates a name of the format interval-n where n is a running number that increments only on un-named intervals
+     *
+     * @param path
+     * @return
+     */
+    public static IntervalList fromVcf(final Path path) {
+        return fromVcf(path, false);
+    }
+
+    public static IntervalList fromVcf(final Path path, final boolean includeFiltered) {
+        final VCFPathReader vcfReader = new VCFPathReader(path, false);
+        final IntervalList intervalList = fromVcf(vcfReader, includeFiltered);
+        vcfReader.close();
+        return intervalList;
+    }
+
+    /**
+     * Converts a vcf to an IntervalList. The name field of the IntervalList is taken from the ID field of the variant, if it exists. If not,
+     * creates a name of the format interval-n where n is a running number that increments only on un-named intervals
+     * Will use a "END" tag in the info field as the end of the interval (if exists).
+     *
+     * @param vcf the vcfReader to be used for the conversion
+     * @return an IntervalList constructed from input vcf
+     */
+
+    public static IntervalList fromVcf(final VCFPathReader vcf) {
+        return fromVcf(vcf, false);
+    }
+
+    public static IntervalList fromVcf(final VCFPathReader vcf, final boolean includeFiltered) {
+
+        //grab the dictionary from the VCF and use it in the IntervalList
+        final SAMSequenceDictionary dict = vcf.getFileHeader().getSequenceDictionary();
+        final SAMFileHeader samFileHeader = new SAMFileHeader();
+        samFileHeader.setSequenceDictionary(dict);
+        final IntervalList list = new IntervalList(samFileHeader);
+
+        int intervals = 0;
+        for (final VariantContext vc : vcf) {
+            if (includeFiltered || !vc.isFiltered()) {
+                String name = vc.getID();
+                final Integer intervalEnd = vc.getCommonInfo().getAttributeAsInt("END", vc.getEnd());
+                if (".".equals(name) || name == null)
```

Comment:
Need `{` `}` around `if` body. Also IMO it's more natural to write `if (name == null || name.equals("."))`